frontend/src/layers/ThQ.ts

```diff
@@ -39,31 +39,30 @@ export class ThQ {
 }
 
 /**
+ * @param thermalVelocity    Thermals velocity in m/s
  * @param boundaryLayerDepth Depth of the boundary layer in meters
  * @param uWind              U part of wind in boundary layer in km/h
  * @param vWind              V part of wind in boundary layer in km/h
- * @param totalCloudCover    Total cloud cover between 0 and 1
- * @returns 
  */
-export const value = (boundaryLayerDepth: number, uWind: number, vWind: number, totalCloudCover: number): number => {
+export const value = (thermalVelocity: number, boundaryLayerDepth: number, uWind: number, vWind: number): number => {
+  // Thermal velocity
+  // coeff is 50% for a 1.5 m/s
+  const thermalVelocityCoeff = logistic(thermalVelocity, 1.50, 6);
+
   // Boundary Layer Depth
-  const bld = boundaryLayerDepth;
   // coeff is 50% for a boundary layer depth of 400 m
-  const bldCoeff = logistic(bld, 400, 4);
+  const bldCoeff = logistic(boundaryLayerDepth, 400, 4);
+
+  const thermalCoeff = (2 * thermalVelocityCoeff + bldCoeff) / 3;
 
   // Boundary Layer Wind
   const u = uWind;
   const v = vWind;
   const windForce = Math.sqrt(u * u + v * v);
-  // coeff is 50% for a wind force of 14 km/h
+  // coeff is 50% for a wind force of 15 km/h
   const windCoeff = 1 - logistic(windForce, 15, 6);
 
-  // Cloud cover
-  const cloudCover = totalCloudCover;
-  // coeff is 50% for a cloud cover of 60%
-  const cloudCoverCoeff = (cloudCover === null || cloudCover === undefined) ? 1 : (1 - logistic(cloudCover, 0.6, 7));
-
-  return bldCoeff * windCoeff * cloudCoverCoeff
+  return thermalCoeff * windCoeff
 };
```

Comment:
The computation of the “XC flying potential” is as follows:

1. We compute a value between 0 and 1 based on the boundary layer depth (0 m = 0.0, 400 m = 0.5, and 800 m = 1.0)
2. We compute a value between 0 and 1 based on the thermals velocity (0 m/s = 0, 1.5 m/s = 0.5, and 3.0 m/s = 1.0)
3. We compute a value between 0 and 1 based on the wind force (0 km/h = 1, 15 km/h = 0.5, and 30 km/h = 0)
4. We multiply the wind coefficient with the average of the boundary layer coefficient and the thermals velocity coefficient

Reply:
interesting, nice: let see how that works.
Could the algorithm be documented in a README somewhere?
Another idea would be to have 2-3 algorithms that could be chosen based on a cookie setting, or url parameter, so you could experiment or a given day when result dont seem to be as expected.